### PR TITLE
Add hosts from host list

### PR DESCRIPTION
**Pull request by flofuchs**
This adds functionality to open a pre-filled form to create new hosts
from discovered hosts from the host list.

### CODE REVIEW

**Comment by andybraren** [comment]
Hi Florian - could you provide screenshots of what this looks like in the hosts list and the form that gets pre-filled when "Add Host" is clicked? I'm curious to see the latest. 🙂

File: src/components/BareMetalHosts/StatusComponents.js [hunk 1]
```diff
@@ -68,8 +68,12 @@ ValidationError.propTypes = {
   errorMessage: PropTypes.string.isRequired,
 };
 
-export const AddDiscoveredHostLink = host => (
-  <a>
+export const AddDiscoveredHostLink = ({ addHostProps }) => (
+  <a onClick={addHostProps.onClick}>
```
Comment (suomiy) [inline src/components/BareMetalHosts/StatusComponents.js]: Are we okay with showing the dialog straight away when clicking on the link ?

Maybe an effort in #427 could be relevant to this.

@andybraren @matthewcarleton @yaacov

Comment (andybraren) [reply src/components/BareMetalHosts/StatusComponents.js]: Definitely relevant. [We have a PR](https://github.com/openshift/openshift-origin-design/pull/179) introducing this new Status Popover pattern in the OpenShift Design repo and it seems likely to get merged.

Here's what a status popover could look like for a discovered host that includes an "Add host" action in the footer. We're still figuring out how discovered hosts should be sorted/separated to make sure they always appear at the top, but if the popover in #427 is easy to implement then something like the below would still be an improvement.

add-host-popover

Comment (flofuchs) [reply src/components/BareMetalHosts/StatusComponents.js]: This should be easy to implement -- let me change the PR to add the popover. I really like it! :)

Comment (flofuchs) [reply src/components/BareMetalHosts/StatusComponents.js]: This is now implemented in this PR

File: src/components/BareMetalHosts/BaremetalHostStatus.js [hunk 2]
```diff
@@ -40,12 +40,14 @@ export const BaremetalHostStatus = ({ host, machine, nodes }) => {
 };
 
 BaremetalHostStatus.propTypes = {
+  addHostProps: PropTypes.object,
```
[the remark this answers is not in the document]
Comment (flofuchs) [reply src/components/BareMetalHosts/BaremetalHostStatus.js]: Done.

File: src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js [hunk 3]
```diff
@@ -117,10 +117,12 @@ export class CreateBaremetalHostDialog extends React.Component {
 
 CreateBaremetalHostDialog.defaultProps = {
   selectedNamespace: null,
+  formValues: {},
```
[the remark this answers is not in the document]
Comment (flofuchs) [reply src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: Done.

File: src/components/BareMetalHosts/fixtures/StatusComponents.fixture.js [hunk 4]
```diff
@@ -62,8 +62,9 @@ export default [
     component: AddDiscoveredHostLink,
     name: 'Show AddDiscoveredHostLink',
     props: {
-      status: 'discovered',
-      text: 'Add host',
+      onAddHost: {
```
Comment (suomiy) [inline src/components/BareMetalHosts/fixtures/StatusComponents.fixture.js]: this prop shouldn't be a object but a function

File: src/components/BareMetalHosts/fixtures/BaremetalHostStatus.fixture.js [hunk 5]
```diff
@@ -20,5 +21,6 @@ export default [
   createFixture('Show progress state for `match profile` status', 'match profile'),
   createFixture('Show error state for `provisioning error` status', 'provisioning error'),
   createFixture('Show error state for `power management error` status', 'power management error'),
-  createFixture('Show add host link for `discovered` status', 'discovered'),
+  createFixture('Show add host link for `discovered` status', 'discovered', { onClick: () => true }),
```
Comment (suomiy) [inline src/components/BareMetalHosts/fixtures/BaremetalHostStatus.fixture.js]: same here

File: src/components/BareMetalHosts/tests/StatusComponents.test.js [hunk 6]
```diff
@@ -61,4 +62,10 @@ describe('<AddDiscoveredHostLink />', () => {
     const component = shallow(testAddDiscoveredHostLink());
     expect(component).toMatchSnapshot();
   });
+
+  it('accepts a callback', () => {
+    const addHostLink = shallow(testAddDiscoveredHostLinkWithCB(() => true));
+    addHostLink.find('a').simulate('click');
+    expect(addHostLink.state('visible')).toBe(true);
```
Comment (suomiy) [inline src/components/BareMetalHosts/tests/StatusComponents.test.js]: shouldn't this be rather false?

**Pull request closed without merging**